drivers/regulator/regulator_pf1550.c

```diff
@@ -0,0 +1,442 @@
+};
+static const struct linear_range ldo2_range[] = {
+	LINEAR_RANGE_INIT(1800000, 100000, 0, 15),
+};
+
+#define PF1550_RAIL_EN        BIT(0)
+#define PF1550_RAIL_EN_MASK   GENMASK(1, 0)
+#define PF1550_GOTO_SHIP      BIT(0)
+#define PF1550_GOTO_SHIP_MASK GENMASK(1, 0)
+
+static const struct regulator_pf1550_desc __maybe_unused buck1_desc = {
+	.vsel_reg = PMIC_SW1_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_SW1_CTRL,
+	.uv_range = buck12_range,
+	.uv_nranges = ARRAY_SIZE(buck12_range),
+	.ua_range = buck123_current_limit_range,
+	.ua_nranges = ARRAY_SIZE(buck123_current_limit_range),
+};
+
+static const struct regulator_pf1550_desc __maybe_unused buck2_desc = {
+	.vsel_reg = PMIC_SW2_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_SW2_CTRL,
+	.uv_range = buck12_range,
+	.uv_nranges = ARRAY_SIZE(buck12_range),
+	.ua_range = buck123_current_limit_range,
+	.ua_nranges = ARRAY_SIZE(buck123_current_limit_range),
+};
+
+static const struct regulator_pf1550_desc __maybe_unused buck3_desc = {
+	.vsel_reg = PMIC_SW3_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_SW3_CTRL,
+	.uv_range = buck3_range,
+	.uv_nranges = ARRAY_SIZE(buck3_range),
+	.ua_range = buck123_current_limit_range,
+	.ua_nranges = ARRAY_SIZE(buck123_current_limit_range),
+};
+
+static const struct regulator_pf1550_desc __maybe_unused ldo1_desc = {
+	.vsel_reg = PMIC_LDO1_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_LDO1_CTRL,
+	.uv_range = ldo13_range,
+	.uv_nranges = ARRAY_SIZE(ldo13_range),
+};
+
+static const struct regulator_pf1550_desc __maybe_unused ldo2_desc = {
+	.vsel_reg = PMIC_LDO2_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_LDO2_CTRL,
+	.uv_range = ldo2_range,
+	.uv_nranges = ARRAY_SIZE(ldo2_range),
+};
+
+static const struct regulator_pf1550_desc __maybe_unused ldo3_desc = {
+	.vsel_reg = PMIC_LDO3_VOLT,
+	.enable_mask = PF1550_RAIL_EN,
+	.enable_val = PF1550_RAIL_EN_MASK,
+	.cfg_reg = PMIC_LDO3_CTRL,
+	.uv_range = ldo13_range,
+	.uv_nranges = ARRAY_SIZE(ldo13_range),
+};
+
+static int regulator_pf1550_set_enable(const struct device *dev, bool enable)
+{
+	const struct regulator_pf1550_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, config->desc->cfg_reg,
+				      config->desc->enable_mask,
+				      enable ? config->desc->enable_val : 0);
+}
+
+static int regulator_pf1550_enable(const struct device *dev)
+{
+	return regulator_pf1550_set_enable(dev, true);
+}
+
+static int regulator_pf1550_disable(const struct device *dev)
+{
+	return regulator_pf1550_set_enable(dev, false);
+}
+
+static int regulator_pf1550_set_mode(const struct device *dev, regulator_mode_t mode)
+{
+	/*
+	 * TODO: support LOAD SWITCH mode
+	 * BIT 4 of LDO1/3_CTRL
+	 */
+	return -ENOTSUP;
+}
```

Comment:
remove, impl is not mandatory (API returns -ENOSYS)

Reply:
I would leave it there so a future contributor can complete the implementation :slightly_smiling_face: